rosetta/validator/block.go

```diff
@@ -54,9 +54,11 @@ func (v *Validator) Block(block identifier.Block) (identifier.Block, error) {
 	}
 	if *block.Index < first {
 		return identifier.Block{}, failure.InvalidBlock{
-			Index:   *block.Index,
-			Hash:    block.Hash,
-			Message: fmt.Sprintf("block index is below first indexed block (first: %d)", first),
+			Index: *block.Index,
+			Hash:  block.Hash,
+			Description: failure.NewDescription("block index is below first indexed height",
```

Comment:
I wanted to nitpick about `first_index`, how `first_height` or `first_indexed` sounds better.

Then I remembered that we're in Rosetta parliance, and `height` does not exist in that realm, and `index` is the correct phrasing.

Then I thought - Rosetta also has no concept of our index either. So actually, `first indexed height` is a bit off, both because of the "height" term and the "indexed" term.

We might just want to say - "block index is below the first known block" or something like that..

Reply:
The _indexed height_ for a block is its _index_, I would argue. Since the Rosetta specification doesn't really specify whether the index is a height or not, I think this makes it clear that the index is based on the height, so I believe it adds value.